qiskit/compiler/schedule.py

```diff
@@ -70,9 +82,19 @@ def schedule(circuits: Union[QuantumCircuit, List[QuantumCircuit]],
         if backend is None:
             raise QiskitError("Must supply either a backend or a meas_map for scheduling passes.")
         meas_map = backend.configuration().meas_map
+    if instruction_durations is None:
+        if backend is None:
+            raise QiskitError("Must supply either a backend or a instruction_durations for "
+                              "scheduling passes.")
```

Comment:
This isn't true. `inst_map` contains this information!

Reply:
This is a point I'd like to discuss more.
Actually, I tried to get durations from `inst_map` at first, i.e. I thought I can implement `InstructionDurations.from_inst_map()` without introducing `instruction_durations` in schedule()'s argument.
However, I had difficulty to get duration from inst_map. The inst_map requires parameters to get Schedule.
We know what parameters are required for predefined gates e.g. u1, u2, u3. So we may use dummy parameters for those three gates.
But what if we find custom gates in inst_map? (How can the program know how many parameters are used in a gate?)
Do you have any good idea?

Reply:
Check this out:
```
schedule(transpiled_circ, my_inst_map, my_meas_map, 'alap')
```
prior to your PR, this would work just fine. It's only sometimes that `instruction_durations` is needed. We don't want to break the previous API by adding this feature.

I don't think `instruction_durations` is required as an input to this function. Let's discuss that in the `sequence.py` file. Here, the minimum is to check if `method = 'sequence'` and then put lines 85-89 and lines 96-97 into that if statement.